src/test/java/co/rsk/federate/FederationProviderFromFederatorSupportTest.java

```diff
@@ -370,9 +371,72 @@ void getRetiringFederation_present_p2sh_erp_federation() {
         assertEquals(expectedFederationAddress, obtainedFederation.getAddress());
     }
 
+    @Test
+    void getProposedFederation_whenRSKIP419IsNotActivated_shouldReturnEmptyOptional() {
+        // Arrange
+        ActivationConfig.ForBlock configMock = mock(ActivationConfig.ForBlock.class);
+        when(configMock.isActive(RSKIP419)).thenReturn(false);
+        when(federatorSupportMock.getConfigForBestBlock()).thenReturn(configMock);
+
+        // Act & Assert
+        assertEquals(Optional.empty(), federationProvider.getProposedFederation());
+    }
+
+    @Test
+    void getProposedFederation_whenProposedFederationSizeIsNonExistent_shouldReturnEmptyOptional() {
+        // Arrange
+        ActivationConfig.ForBlock configMock = mock(ActivationConfig.ForBlock.class);
+        when(configMock.isActive(RSKIP419)).thenReturn(true);
+        when(federatorSupportMock.getConfigForBestBlock()).thenReturn(configMock);
+        when(federatorSupportMock.getProposedFederationSize())
+            .thenReturn(Optional.of(FEDERATION_NON_EXISTENT.getCode()));
+
+        // Act & Assert
+        assertEquals(Optional.empty(), federationProvider.getProposedFederation());
+        verify(federatorSupportMock).getProposedFederationSize();
+    }
+
+    @Test
+    void getProposedFederation_whenExistsAndIsP2shErpFederation_shouldReturnProposedFederation() {
+        // Arrange
+        ActivationConfig.ForBlock configMock = mock(ActivationConfig.ForBlock.class);
+        when(configMock.isActive(RSKIP419)).thenReturn(true);
+        when(federatorSupportMock.getConfigForBestBlock()).thenReturn(configMock);
+
+        Federation expectedFederation = createP2shErpFederation(
+            getFederationMembersFromPks(1, 1000, 2000, 3000, 4000, 5000));
+        Address expectedFederationAddress = expectedFederation.getAddress();
+        Integer federationSize = 5;
+        when(federatorSupportMock.getProposedFederationSize()).thenReturn(Optional.of(federationSize));
+        when(federatorSupportMock.getProposedFederationCreationTime()).thenReturn(Optional.of(creationTime));
+        when(federatorSupportMock.getProposedFederationAddress()).thenReturn(Optional.of(expectedFederationAddress));
+        when(federatorSupportMock.getBtcParams()).thenReturn(testnetParams);
+        when(federatorSupportMock.getProposedFederationCreationBlockNumber()).thenReturn(Optional.of(0L));
+        for (int i = 0; i < federationSize; i++) {
+            when(federatorSupportMock.getProposedFederatorPublicKeyOfType(i, FederationMember.KeyType.BTC))
```

Comment:
Test case when the key does not exist and it throws an `IllegalStateException`, wdyt?

Reply:
Updated